config/http_config.go

```diff
@@ -501,6 +527,13 @@ func NewClientFromConfig(cfg HTTPClientConfig, name string, optFuncs ...HTTPClie
 // given config.HTTPClientConfig and config.HTTPClientOption.
 // The name is used as go-conntrack metric label.
 func NewRoundTripperFromConfig(cfg HTTPClientConfig, name string, optFuncs ...HTTPClientOption) (http.RoundTripper, error) {
+	return NewRoundTripperFromConfigWithContext(context.Background(), cfg, name, optFuncs...)
+}
+
+// NewRoundTripperFromConfig returns a new HTTP RoundTripper configured for the
+// given config.HTTPClientConfig and config.HTTPClientOption.
```

Comment:
lmao, what happened that we have both *Config and *Option 🙈

Reply:
The config is the YAML configuration. We must make the fields public for `prometheus/prometheus` to be able to set the values. However, once you do that, `yaml` will start to unmarshal those fields, which we do not want. Hence why there is a separate object for setting additional configuration outside of the user-specified configuration. This is a limitation of the `yaml` library.

Reply:
Makes sense